keyboards/m65/keymaps/default/keymap.c

```diff
@@ -0,0 +1,418 @@
+#endif
+  }
+
+}
+
+static inline void led_cpp(const bool on){
+
+  if (on){
+#ifdef KEYBOARD_m65_rev1
+    writePinHigh(B12);
+#endif
+  } else {
+#ifdef KEYBOARD_m65_rev1
+    writePinLow(B12);
+#endif
+  }
+
+}
+
+void matrix_init_user(void) {
+#ifdef KEYBOARD_m65_rev1
+  setPinOutput(B13);
+  setPinOutput(B12);
+#endif
+}
+
+void matrix_scan_user(void) {
+
+  led_fortran(IS_LAYER_ON(_Fortran));
+  led_cpp(IS_LAYER_ON(_CPP));
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  // bool lalt = (keyboard_report->mods & MOD_BIT(KC_LALT));
+  switch (keycode) {
+    case M_EXTDISP:
+      if (record->event.pressed) {
+        TAP(LGUI(KC_P));
+      }
+      break;
+    case M_FIND:
+      if (record->event.pressed) {
+        TAP(LALT(KC_F2));
+      }
+      break;
+    case F_Q:
+      if (record->event.pressed) {
+        SEND_STRING("intent(inout)");
+      }
+      break;
+    case F_W:
+      if (record->event.pressed) {
+        SEND_STRING("write()");
+        TAP(KC_LEFT);
+      }
+      break;
+    case F_R:
+      if (record->event.pressed) {
+        SEND_STRING("read()");TAP(KC_LEFT);
+      }
+      break;
+    case F_E:
+      if (record->event.pressed) {
+        SEND_STRING("exp()");TAP(KC_LEFT);
+      }
+      break;
+    case F_T:
+      if (record->event.pressed) {
+        SEND_STRING("type :: ");TAP(KC_ENT);
+        SEND_STRING("end type");TAP(KC_UP);
+      }
+      break;
+    case F_Y:
+      if (record->event.pressed) {
+        SEND_STRING("call random_number()");TAP(KC_LEFT);
+      }
+      break;
+    case F_U:
+      if (record->event.pressed) {
+        SEND_STRING("use ");
+      }
+      break;
+    case F_I:
+      if (record->event.pressed) {
+        SEND_STRING("if () then");
+        TAP(KC_ENT);
+        SEND_STRING("else");
+        TAP(KC_ENT);
+        SEND_STRING("end if");
+        TAP(KC_UP);
+        TAP(KC_UP);
+        TAP(KC_LEFT);
+        TAP(KC_LEFT);
+      }
+      break;
+    case F_O:
+      if (record->event.pressed) {
+        SEND_STRING("open()");TAP(KC_LEFT);
+      }
+      break;
+    case F_P:
+      if (record->event.pressed) {
+        SEND_STRING("parameter");
+      }
+      break;
+    case F_A:
+      if (record->event.pressed) {
+        SEND_STRING("return");TAP(KC_ENT);
+      }
+      break;
+    case F_S:
+      if (record->event.pressed) {
+        SEND_STRING("subroutine");TAP(KC_ENT);
+        SEND_STRING("end subroutine");TAP(KC_UP);
+      }
+      break;
+    case F_D:
+      if (record->event.pressed) {
+        SEND_STRING("do ");
+        TAP(KC_ENT);
+        SEND_STRING("end do");
+        TAP(KC_UP);
+      }
+      break;
+    case F_F:
+      if (record->event.pressed) {
+        SEND_STRING("function");TAP(KC_ENT);
+        SEND_STRING("end function");TAP(KC_UP);
+      }
+      break;
+    case F_G:
+      if (record->event.pressed) {
+        SEND_STRING("https://en.wikipedia.org/wiki/Goto");
+      }
+      break;
```

Comment:
This is cute, although I can't help but feel that you should configure your editor to do this, rather than build it into your keyboard.

In any case, you should swap the nesting of `switch` and `if`, and also omit the empty `case …: break;` clauses entirely for much shorter & maintainable code:
```
if (!record->event.pressed) return true;
switch (keycode) {
    case F_G:
        SEND_STRING("https://en.wikipedia.org/wiki/Goto");
        break;
}
```

Reply:
i end up working on silly editors and I would like not to configure each of them... especially since I work in cli on various remote machines... read vi or vim

Reply:
But Vim is highly configurable! Plus it has a good idea what language language you're working in—your keyboard doesn't. It's easy enough to add per-language shortcuts with e.g.:
```
$ cat ~/.vim/ftplugin/fortran.vim 
abbreviate <buffer> FF for (i = 0; i < ; ++i)
```
(I just copied that example out of Vim's `:help abbreviate-local`.)

You can keep a copy of your Vim config in a `git` repo and just clone them to any machine you happen to be working on. You can't easily `scp` nor `git clone` your keyboard.

But if you're stuck with `ed` or `nano` well…

Reply:
i know but when you use 10 machines you really do not want all that in the keyboard... no worries I understand people do not really use these things. I think I will clean them from there maybe I let only one or two for reference